### ISSUE TITLE
Steam Deck UI issues

### ISSUE THREAD

**Issue by LukeShortCloud** [LukeShortCloud/winesapOS#277]
I have to remove -gamepadui to sign in cause you won't able to setup steamdeck UI and performance was very buggy only got 23 fps on apex legends compare to older version which gave me 80 to 117fps

_Originally posted by @massatt212 in https://github.com/LukeShortCloud/winesapOS/issues/272#issuecomment-1072505223_

**Comment by LukeShortCloud** [comment]
What do you mean by "you won't be able to setup steamdeck UI"? I was able to login and use it on my AMD dGPU and my Intel integrated graphics. On my NVIDIA dGPU, it didn't work at all (it would crash). What GPU as you using? I expect it won't work the best on anything except AMD GPUs right now.

Did you also happen to go through the first-time setup to configure the graphics driver? There's a new bug we found where if winesapOS does not have Internet connection upon your first login then that will crash / not load. There's a desktop shortcut for it you can use to launch it again, if needed: "winesapOS First-Time Setup".

Also, just to clarify, did the performance of Apex improve when removing the "-gamepadui" argument and then launching Steam normally?

**Comment by massatt212** [comment]
I have a Rx 470 gb and I did went through first time setup selected amd I couldn't setup the first tike setup for steam and the performance was really bad, I'll try it again on the next update.

**Comment by massatt212** [comment]
I already removed it, I'll  get an external and test it out ltr.

**Comment by massatt212** [comment]
is there a way to skip steam ui first time setup, I'm using -steamos3 -steampal -steamdeck to get fsr and mangohud using quick access menu and shortcut steam button + (A), is their a command something like -nointro to get pass first-time setup.

**Comment by LukeShortCloud** [comment]
Are you trying to skip the winesapOS First-Time Setup or something with Steam? For winesapOS, you need to delete the symlink file `/home/winesap/.config/autostart/winesapos-setup.desktop`. Upon first boot up, switch to a different TTY, log in as root first, delete the file, switch back to the GUI TTY, and then login. Not exactly easy but it's doable.

I also didn't know about those other Steam client arguments. That's neat! I'll play around with that, too. I plan on having two desktop shortcuts for Steam in our stable release: "Steam (Runtime)" and "Steam Deck (Runtime)".

By the way, are you still experiencing any drops in gaming performance compared to before?

**Comment by massatt212** [comment]
the performance drop was from my side, gpu needed to be reseated

**Comment by massatt212** [comment]
if possible you should try make a version of steamdeck os, their are lots of people who want to try it out, but the system update at the first time setup is stopping a lot of people.

**Comment by massatt212** [comment]
#!/bin/sh
export SDL_VIDEO_MINIMIZE_ON_FOCUS_LOSS=0

export HOMETEST_DESKTOP=1
export HOMETEST_USER=steam
export HOMETEST_DESKTOP_USER=desktop
export HOMETEST_DESKTOP_SESSION=gnome

# Add our bin directory with the set_hd_mode and dpkg-query replacement scripts
export PATH=/usr/share/steamos-compositor-plus/bin:${PATH}

set_hd_mode.sh >> $HOME/.set_hd_mode.log 2>&1

export LD_PRELOAD=/usr/lib/libmodeswitch_inhibitor.so:/usr/lib32/libmodeswitch_inhibitor.so

# Disable DPMS and screen blanking for now; it doesn't know about controller
# or Steam remote control events right now
xset -dpms
xset s off

if command -v start-pulseaudio-x11 > /dev/null; then
	start-pulseaudio-x11 &
fi

if systemctl is-active --quiet bluetooth && command -v bluetoothctl > /dev/null; then
	bluetoothctl power on
fi

CONFIG_PATH=${XDG_CONFIG_HOME:-$HOME/.config}
CONFIG_FILE="$CONFIG_PATH/steamos-compositor-plus"

STEAMCMD="/usr/bin/gamescope -U -e -f -h 1080 -- steam -gamepadui -steamos -pipewire-dmabuf -tenfoot"
COMPOSITORCMD="steamcompmgr"

if command -v xdg-settings > /dev/null; then
	if [ ! -f ${CONFIG_PATH}/mimeapps.list ]; then
		xdg-settings set default-web-browser steam-browser.desktop
	fi
fi

# Source environment from ~/.config/environment.d
for i in ~/.config/environment.d/*.conf ;
do
    if [ -f "${i}" ]; then
        set -a
        source "${i}"
        set +a
    fi
done

# Source our config file
if [ -f "$CONFIG_FILE" ]; then
	source "$CONFIG_FILE"
fi

$COMPOSITORCMD &

loadargb_cursor /usr/share/icons/steam/arrow.png

# Workaround for Steam login issue while Steam client change propagates out of Beta
touch ~/.steam/root/config/SteamAppData.vdf || true

# Workaround for missing background when starting steam in big picture mode for the first time
DATA_HOME=${XDG_DATA_HOME:-$HOME/.local/share}
IMAGES_DIR="$DATA_HOME/Steam/tenfoot/resource/images"
BOOTSTRAP_IMAGE="$IMAGES_DIR/bootstrapper.jpg"
if [ ! -f "$BOOTSTRAP_IMAGE" ]; then
	mkdir -p "$IMAGES_DIR"
	cp /usr/share/pixmaps/steam-bootstrapper.jpg "$BOOTSTRAP_IMAGE"
fi

if command -v steam-tweaks > /dev/null; then
	steam-tweaks
fi

$STEAMCMD

**Comment by massatt212** [comment]
thats steamos compositor plus running steamdeck ui
i cant get -steamos3 in 1080p and i cant bypass the -steamdeck for fsr and gamescope support, maybe you can check it out and see what you can do.

**Comment by massatt212** [comment]
The first time setup for steam Deck UI not the Linux system for example boot this command in the terminal and try setup steam Deck UI

Steam -gamepadui -steampal -steamo3 -steamdeck

Run that in terminal and see what I'm talking about, cause once you never login steam the -gamepadui will boot and prevent you from logging in, that command will give you an example of what I'm talking About.

**Comment by LukeShortCloud** [comment]
Thanks for the clarification, I appreciate that. I'll try after I'm done with work today.

**Comment by LukeShortCloud** [comment]
My Intel HD 5500 iGPU used to load up the Steam Deck client but now is stuck on a screen with the Steam Deck logo. I tried every combination of CLI arguments for Steam and ran it from the terminal to watch the debug output. There were a lot of OpenGL errors. I even tried switching my Intel OpenGL graphics driver over to using Zink but that lead to major graphics issues (but I could still see it only got as far as the Steam Deck logo).

The only hardware I can get to reliably run the Steam Deck client is my computer with an AMD Radeon RX 6300M. Considering that the Steam Deck uses a modern AMD GPU, that is likely the only hardware Valve is targeting for support right now.

Unfortunately, this is out of our hands. We need to wait for Valve to roll-out the new user interface to desktop users (something they said will happen one day, at least). I'll still keep the Steam Deck shortcut on the desktop (there's now a "Steam Desktop" and separate "Steam Deck" shortcut) and add a disclaimer to our documentation.

I'll continue to keep a look out on Twitter, Reddit, forums, etc. for any workarounds in the meantime.

**Comment by massatt212** [comment]
I want to get the steam Deck quick access menu without the series s or ps5 controller and the resolution is quit messed up running -steamo3, but nice to see you made some progress.

**Comment by LukeShortCloud** [comment]
I was able to get the Steam Deck client working on my NVIDIA computer! Here is the summary of what worked for me:

(1) Used the NVIDIA 495 driver from SteamOS (downgraded from 510 that I was using on Manjaro). There are no longer graphical artifacts. I know you have an AMD dGPU so this isn't exactly related to your problem.
(2) Relevant to you, though, I launched the normal "Steam Desktop" client, signed in (and selected to stay signed in), and launched a game. Then I closed out of that completely and was then able to launch the "Steam Deck" client. Since I was already logged in, it launched and gave me a quick 5 second tutorial of the Steam Deck UI, and then I was able to access the UI, browse my library, and launched a game.

I'll note this in our stable release notes tomorrow. Hopefully this helps you.
